cmd/cli/user.go

```diff
@@ -49,10 +59,79 @@ func deleteUser(database moira.Database, user string) error {
 	if err != nil {
 		return err
 	}
+
 	for _, contactID := range contactIDs {
 		if err := database.RemoveContact(contactID); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
+
+func handleCleanup(database moira.Database, conf cleanupConfig) error {
+	var users []string
+
+	reader := json.NewDecoder(bufio.NewReader(os.Stdin))
+
+	if err := reader.Decode(&users); err != nil {
+		return err
+	}
+
+	return usersCleanup(database, users, conf)
+}
+
+func usersCleanup(database moira.Database, users []string, config cleanupConfig) error {
+	whitelist := make(map[string]bool)
```

Comment:
It is necessary to check that `config.Whitelist` is not nil. Here is PoC for bug https://play.golang.org/p/nIxb2jNdoZw

Reply:
I don’t think we should think about this. We are waiting for a list, not a string line.
```data2 := []byte("hello: world")```

Reply:
This string represents yaml with invalid structure and missing whitelist field. This issue is still need fix